### PR TITLE
feat(textures): Add texture files

### PR DESCRIPTION
**Pull request by hellmor**
feats(textures): Add texture files

Add texture2D and textureCube files into engine.
----
Added: TextureCube.ts
Added: TextureMipmapCompute.ts
Added: TextureMipmapGenerator.ts
Added: AtmosphericScatteringSky.ts
Added: BitmapTexture2D.ts
Added: BitmapTextureCube.ts
Added: DefaultRes.ts
Added: Float16ArrayTexture.ts
Added: Float32ArrayTexture.ts
Added: HDRTexture.ts
Added: HDRTextureCube.ts
Added: LDRTextureCube.ts
Added: SolidColorSky.ts
Added: Uint16Texture.ts
Added: Uint8ArrayTexture.ts

### CODE REVIEW

File: src/engine/gfx/graphics/generate/convert/ErpImage2CubeMap.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        GPUContext.endCommandEncoder(commandEncoder);
+
+        configBuffer.destroy();
+    }
+
+    private static makeFaceTexturePipeline: GPUComputePipeline;
+    private static configBuffer: GPUBuffer;
+    private static quaternionBuffer: GPUBuffer;
+
+    //image为从rgbe转换为rgba的float32颜色值
```
Comment (Codeboy-cn) [inline src/engine/gfx/graphics/generate/convert/ErpImage2CubeMap.ts:83]: Please use english or remove it.

File: src/engine/gfx/graphics/webGpu/core/texture/TextureCube.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import { GPUAddressMode } from '../../WebGPUConst';
+import { Texture } from './Texture';
+/**
+ * 立方体纹理
```
Comment (Codeboy-cn) [inline src/engine/gfx/graphics/webGpu/core/texture/TextureCube.ts:4]: Please use english or remove it.

File: src/engine/gfx/graphics/webGpu/core/texture/TextureMipmapGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        if (!pipeline) {
+            mipmapShaderModule = device.createShaderModule({
+                code: TextureMipmapGenerator.mipmapShader,
+            });
+
+            let textureLayout = device.createBindGroupLayout({
+                entries: [
+                    {
+                        binding: 0,
+                        //TODO : 加入shader reflection 之后需要知道 顶点用到 | 片段用到
```
Comment (Codeboy-cn) [inline src/engine/gfx/graphics/webGpu/core/texture/TextureMipmapGenerator.ts:47]: Please use english or remove it.

File: src/engine/gfx/graphics/webGpu/core/texture/TextureMipmapGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+                        visibility: texture.visibility,
+                        // use GPUSamplerBindingLayout = { type:`filtering`} error
+                        sampler: texture.samplerBindingLayout,
+                    },
+                    {
+                        binding: 1,
+                        //TODO : 加入shader reflection 之后需要知道 顶点用到 | 片段用到
```
Comment (Codeboy-cn) [inline src/engine/gfx/graphics/webGpu/core/texture/TextureMipmapGenerator.ts:54]: Please use english or remove it.

File: src/engine/textures/HDRTextureCube.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    public uploadTexture(mip: number, texture: Texture): this {
+        let gpuSource = this.getGpuSource(mip);
+        ErpImage2CubeMap.makeTextureCube(texture, this.width, gpuSource.v);
+        return this;
+    }
+
+    /**
+     * 获取GPU纹理原始数据
+     * @param mip Mipmap级数
+     * @returns GPU纹理原始数据，包含t: GPUTexture与v: GPUTextureView
```
Comment (Codeboy-cn) [inline src/engine/textures/HDRTextureCube.ts:99]: Please use english.

File: src/engine/textures/HDRTextureCube.ts [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                }),
+            };
+            this.faceTextureRef[mip] = source;
+        }
+        return source;
+    }
+
+    /**
+     * 生成Mipmap
+     * @param texture 纹理对象
```
Comment (Codeboy-cn) [inline src/engine/textures/HDRTextureCube.ts:121]: Please use english.

File: src/engine/textures/HDRTextureCube.ts [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            this.generateMipmapAtLevel(mipmap, texture);
+            mipmap++;
+        }
+    }
+
+    /**
+     * 生成指定级别的Mipmap
+     * @param mipmap Mipmap级别
+     * @param erpTexture erp纹理对象
+     * @param pow 指数系数
```
Comment (Codeboy-cn) [inline src/engine/textures/HDRTextureCube.ts:135]: Please use english.

File: src/engine/textures/LDRTextureCube.ts [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        this.gpuSampler = webGPUContext.device.createSampler(this);
+
+        this.uploadErpTexture(texture);
+        return this;
+    }
+
+    /**
+     * @private
+     * @param texture 纹理对象
+     * @returns 纹理立方体对象
```
Comment (Codeboy-cn) [inline src/engine/textures/LDRTextureCube.ts:88]: Please use english.

File: src/engine/textures/LDRTextureCube.ts [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        let gpuSource = this.getGpuSource(0);
+        ErpImage2CubeMap.makeTextureCube(texture, this.width, gpuSource.v);
+        this.generateMipmap(texture);
+        return this;
+    }
+
+    /**
+     * @priate 获取GPU纹理原始数据
+     * @param mip Mipmap级数
+     * @returns GPU纹理原始数据，包含t: GPUTexture与v: GPUTextureView
```
Comment (Codeboy-cn) [inline src/engine/textures/LDRTextureCube.ts:100]: Please use english.

File: src/engine/textures/LDRTextureCube.ts [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+                }),
+            };
+            this._faceTextureRef[mip] = source;
+        }
+        return source;
+    }
+
+    /**
+     * @private 生成Mipmap
+     * @param texture 纹理对象
```
Comment (Codeboy-cn) [inline src/engine/textures/LDRTextureCube.ts:122]: Please use english.

File: src/engine/textures/LDRTextureCube.ts [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+            this.generateMipmapAtLevel(mipmap, texture);
+            mipmap++;
+        }
+    }
+
+    /**
+     * @private 生成指定级别的Mipmap
+     * @param mipmap Mipmap级别
+     * @param erpTexture erp纹理对象
+     * @param pow 指数系数
```
Comment (Codeboy-cn) [inline src/engine/textures/LDRTextureCube.ts:136]: Please use english.

File: src/engine/gfx/graphics/generate/convert/ErpImage2CubeMap.ts (hunk 1 above)
Comment (hellmor) [reply src/engine/gfx/graphics/generate/convert/ErpImage2CubeMap.ts:83]: fixed

File: src/engine/gfx/graphics/webGpu/core/texture/TextureCube.ts (hunk 2 above)
Comment (hellmor) [reply src/engine/gfx/graphics/webGpu/core/texture/TextureCube.ts]: fixed

File: src/engine/gfx/graphics/webGpu/core/texture/TextureMipmapGenerator.ts (hunk 4 above)
Comment (hellmor) [reply src/engine/gfx/graphics/webGpu/core/texture/TextureMipmapGenerator.ts]: fixed

File: src/engine/textures/LDRTextureCube.ts (hunk 11 above)
Comment (hellmor) [reply src/engine/textures/LDRTextureCube.ts]: fixed.

File: src/engine/textures/LDRTextureCube.ts (hunk 10 above)
Comment (hellmor) [reply src/engine/textures/LDRTextureCube.ts]: fixed.

File: src/engine/textures/LDRTextureCube.ts (hunk 9 above)
Comment (hellmor) [reply src/engine/textures/LDRTextureCube.ts]: fixed.

File: src/engine/textures/LDRTextureCube.ts (hunk 8 above)
Comment (hellmor) [reply src/engine/textures/LDRTextureCube.ts]: fixed.

File: src/engine/textures/HDRTextureCube.ts (hunk 7 above)
Comment (hellmor) [reply src/engine/textures/HDRTextureCube.ts]: fixed.

**Pull request closed without merging**